### ISSUE TITLE
Cannot build Prismatik from source

### ISSUE THREAD

**Issue by dmadison** [psieg/Lightpack#213]
I'm trying to compile Prismatik from source and running into a couple of issues. I've successfully installed and configured Qt and generated the Prismatik solution, but it's failing to compile.

First, it seems like bass support cannot be made optional. I have the three BASS lines commented out in `build-vars.prf`, but Visual Studio throws an error that it's missing `bass.h`:
```
3>c:\users\dave\documents\github\lightpack\software\src\soundmanager.cpp(31): fatal error C1083: Cannot open include file: 'bass.h': No such file or directory
```

Second, it looks like it's having issues with at least one of the auto-generated Qt files:
```
3>c:\users\dave\documents\github\lightpack\software\src\stuff\moc_systrayicon_qt_p.cpp(9): fatal error C1083: Cannot open include file: '../systrayicon/SysTrayIcon_qt_p.hpp': No such file or directory
```

Any idea what I'm doing wrong? I'm using Visual Studio 2017 with Qt 5.12.0 (msvc2017_64) on Windows 10 x64.

**Comment by zomfg** [comment]
You could try removing `BASS_SOUND_SUPPORT` from projects' preprocessor definitions.

Maybe your Qt is too new? I'm using 5.7.1 and `moc_*` files don't have `_qt_p` suffixes.

**Comment by dmadison** [comment]
Looks like removing `BASS_SOUND_SUPPORT` from the project's preprocessor definitions still results in the same error, and seems to cause a lot of other errors related to the sound visualizer.

I'll try installing an older version of Qt.

**Comment by dmadison** [comment]
I've cleaned the repo (`git clean -f -d -x`) and tried again. With Qt 5.7.1 the batch file won't even build the solution, and the project file won't open in Qt Creator either. Seems it's missing the serial component:

```
C:\Users\Dave\Documents\GitHub\Lightpack\Software>D:\Qt\Qt5.7.1\5.7\winrt_x64_msvc2015\bin\qmake -recursive -tp vc Lightpack.pro
Reading C:/Users/Dave/Documents/GitHub/Lightpack/Software/src/src.pro
WARNING: Project ERROR: Unknown module(s) in QT: serialport
Updating 'C:/Users/Dave/Documents/GitHub/Lightpack/Software/src/../res/translations/ru_RU.qm'...
    Generated 151 translation(s) (150 finished and 1 unfinished)
    Ignored 24 untranslated source text(s)
Updating 'C:/Users/Dave/Documents/GitHub/Lightpack/Software/src/../res/translations/uk_UA.qm'...
    Generated 96 translation(s) (78 finished and 18 unfinished)
    Ignored 79 untranslated source text(s)
Project ERROR: Unknown module(s) in QT: serialport
```
Some cursory searching on Google says that the `serialport` module is included in the Qt installer for version 5 and can't be added separately. Which would suggest that the project is not configured for this version of Qt...

I'm going to try installing the version of Qt for when 7a80d18 was pushed (should be 5.11.2?), and doing this again with a freshly cloned copy of the repo.

**Comment by zomfg** [comment]
I just cloned a fresh `master`, copied `build-vars.prf`, commented all `BASS` related stuff, generated the solution with the `.bat`, opened in VS2017 said yes to all upgrades to projects, and was able to build on first try.
That's still with Qt 5.7.1, with 5.12 I can't even generate the solution for whatever reason.

One problem I have with VS2017 is I can't launch performance analysis, but I don't think it's directly related to Prismatik.

**Comment by dmadison** [comment]
Okay, I have it compiled and working. Here's what I did:

1. Cloned a new repository from `master`.
2. Downloaded and installed [Qt 5.11.2](https://download.qt.io/archive/qt/5.11/5.11.2/) for Windows x86, released 9/18/18.
3. Set a system variable of `QTDIR` to the directory of the installation (D:\Qt\5.11.2\msvc2017_64).
4. Followed the instructions on the README:

```
1. Go to <repo>/Software
2. Copy and edit build-vars.prf according to your machine
3. Run scripts/win32/generate_sln.bat (from the Visual Studio Developer prompt / vcvarsall.bat)
4. Build Lightpack.sln with MSBuild / VisualStudio
```

For `build-vars.prf` (item 2) I commented out the sound support lines and left `DEFINES += NO_OPENSSL` uncommented.

My guess is that when I did my initial pull to update my local repo I didn't clean the `gitignore`'d files, so they were interfering with the build process.

It may be prudent to look into why the solution isn't building with the latest version of Qt, but that's above my pay grade.
